pyspark/bigdl/ml_piepeline/dl_classifier.py

```diff
@@ -0,0 +1,118 @@
+from pyspark.ml.pipeline import Estimator, Model
+from pyspark.ml.param.shared import *
+from bigdl.util.common import *
+from pyspark.ml.wrapper import JavaParams
+
+if sys.version >= '3':
+    long = int
+    unicode = str
+
+
+class HasBatchSize(Params):
+    """
+    Mixin for param batchSize: batch size.
+    """
+
+    # a placeholder to make it appear in the generated doc
+    batchSize = Param(Params._dummy(), "batchSize", "batchSize")
+
+    def __init__(self):
+        super(HasBatchSize, self).__init__()
+        #: param for batch size.
+        self.batchSize = Param(self, "batchSize", "batchSize")
+        self._setDefault(batchSize=1)
+
+    def setBatchSize(self, value):
+        """
+        Sets the value of :py:attr:`batchSize`.
+        """
+        self._paramMap[self.batchSize] = value
+        return self
+
+    def getBatchSize(self):
+        """
+        Gets the value of batchSize or its default value.
+        """
+        return self.getOrDefault(self.batchSize)
+
+
+class DLEstimator(JavaParams, Estimator, HasFeaturesCol, HasLabelCol, HasPredictionCol, HasBatchSize):
```

Comment:
Can we make it a subclass of JavaValue too? If so, there is no need to duplicate the jvm_class_constructor code below.

Reply:
I delete the` jvm_class_constructor` method declaration, import `JavaValue` from `bigdl.util.common` and replace it with JavaValue.jvm_class_constructor now. JavaValue's init method needs to specify parameters DLEstimator doesn't have like `jvalue` which could be easier by just using the class method outside the JavaValue class.

Reply:
that works for now

Reply:
You can put `None` for `jvalue` if its empty.
The code is duplicate if without inheritance:
```
+    def jvm_class_constructor(self):
+        name = "create" + self.__class__.__name__
+        print("creating: " + name)
+        return name
+
+    def __init__(self,  model, criterion, featureSize, labelSize, bigdl_type="float"):
+        super(DLEstimator, self).__init__()
+        self.value = callBigDlFunc(
+            bigdl_type, self.jvm_class_constructor(), model, criterion, featureSize, labelSize)
+        self.bigdl_type = bigdl_type
```

```

Reply:
already fixed